.github/workflows/testing.yml

```diff
@@ -232,6 +232,8 @@ jobs:
       - name: Download artifacts
         uses: actions/download-artifact@v2
       - name: Upload to CodeCov
-        uses: codecov/codecov-action@v2
+        uses: codecov/codecov-action@v3
         with:
           fail_ci_if_error: true
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
will see if this breaks when a pull request is opened from a fork instead of the original repo.